Formula/n/nvtop.rb

```diff
@@ -0,0 +1,21 @@
+class Nvtop < Formula
+  desc "Interactive GPU process monitor"
+  homepage "https://github.com/Syllo/nvtop"
+  url "https://sid.ethz.ch/debian/nvtop/macos/nvtop_3.0.2%2Bgit20240213%2Bds.orig.tar.xz"
+  sha256 "fb46aa755eeeadab6e6f511915ab0166f813fb0c1d2dd929bf3a3249b6ae10d0"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkgconf" => :build
+
+  def install
+    system "cmake", "."
+    system "make"
+    bin.install "src/nvtop"
+    man1.install "manpage/nvtop" => "nvtop.1"
+  end
+
+  test do
+    system "#{bin}/nvtop", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.